testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/KafkaSsl.java

```diff
@@ -0,0 +1,118 @@
+ * limitations under the License.
+ */
+
+package io.trino.tests.product.launcher.env.common;
+
+import io.trino.tests.product.launcher.env.DockerContainer;
+import io.trino.tests.product.launcher.env.Environment;
+import io.trino.tests.product.launcher.testcontainers.PortBinder;
+import org.testcontainers.containers.BindMode;
+import org.testcontainers.containers.startupcheck.IsRunningStartupCheckStrategy;
+
+import javax.inject.Inject;
+
+import static io.trino.tests.product.launcher.docker.ContainerUtil.forSelectedPorts;
+import static java.util.Objects.requireNonNull;
+import static org.testcontainers.containers.wait.strategy.Wait.forLogMessage;
+
+public class KafkaSsl
+        implements EnvironmentExtender
+{
+    private static final String CONFLUENT_VERSION = "5.5.2";
+    private static final int SCHEMA_REGISTRY_PORT = 8081;
+
+    private final PortBinder portBinder;
+
+    @Inject
+    public KafkaSsl(PortBinder portBinder)
+    {
+        this.portBinder = requireNonNull(portBinder, "portBinder is null");
+    }
+
+    @Override
+    public void extendEnvironment(Environment.Builder builder)
+    {
+        builder.addContainers(createZookeeper(), createKafka(), createSchemaRegistry())
+                .containerDependsOn("kafka", "zookeeper")
+                .containerDependsOn("schema-registry", "kafka");
+    }
+
+    @SuppressWarnings("resource")
+    private DockerContainer createZookeeper()
+    {
+        DockerContainer container = new DockerContainer("confluentinc/cp-zookeeper:" + CONFLUENT_VERSION, "zookeeper")
+                .withEnv("ZOOKEEPER_CLIENT_PORT", "2181")
+                .withEnv("ZOOKEEPER_TICK_TIME", "2000")
+                .withStartupCheckStrategy(new IsRunningStartupCheckStrategy())
+                .waitingFor(forSelectedPorts(2181));
+
+        portBinder.exposePort(container, 2181);
+
+        return container;
+    }
+
+    @SuppressWarnings("resource")
+    private DockerContainer createKafka()
+    {
+        DockerContainer container = new DockerContainer("confluentinc/cp-kafka:" + CONFLUENT_VERSION, "kafka")
+                .withEnv("KAFKA_BROKER_ID", "1")
+                .withEnv("KAFKA_ZOOKEEPER_CONNECT", "zookeeper:2181")
+                .withEnv("KAFKA_ADVERTISED_LISTENERS", "SSL://kafka:9092")
+                .withEnv("KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR", "1")
+                .withEnv("KAFKA_GROUP_INITIAL_REBALANCE_DELAY_MS", "0")
+                .withEnv("KAFKA_SSL_KEYSTORE_FILENAME", "kafka.broker1.keystore.jks")
+                .withEnv("KAFKA_SSL_KEYSTORE_CREDENTIALS", "broker1_keystore_creds")
+                .withEnv("KAFKA_SSL_KEY_CREDENTIALS", "broker1_sslkey_creds")
+                .withEnv("KAFKA_SSL_TRUSTSTORE_FILENAME", "kafka.broker1.truststore.jks")
+                .withEnv("KAFKA_SSL_TRUSTSTORE_CREDENTIALS", "broker1_truststore_creds")
+                .withEnv("KAFKA_SSL_ENDPOINT_IDENTIFICATION_ALGORITHM", " ")
```

Comment:
Why not `HTTPS`?

Reply:
Good question. It was done this way previously by @MStrobl . I'll add HTTPS in this case!

Reply:
Sorry, I forgot setting it.